src/index.ts

```diff
@@ -1,18 +1,101 @@
-import MeiliSearch from 'meilisearch'
-import { removeUndefinedFromObject } from './utils.js'
-import { createHighlighResult, createSnippetResult } from './format.js'
+import MeiliSearch, { SearchParams, SearchResponse } from 'meilisearch'
 
-export default function instantMeiliSearch(hostUrl, apiKey, options = {}) {
+import { createHighlighResult, createSnippetResult } from './format'
+import { removeUndefinedFromObject } from './utils'
+
+export type Params = {
+  page?: number
+  hitsPerPage?: number
+  highlightPreTag?: string
+  highlightPostTag?: string
+  snippetEllipsisText?: string
+  attributesToSnippet?: string[]
+  query: string
+  facets: string[] // TODO: exclude key facetsDistribution because this one is used instead
+} & SearchParams<any>
+
+type Request = {
+  params: Params
+  indexName: string
+}
+type Requests = {
+  0: Request
+}
+type InstantMeiliSearchOptions = {
+  paginationTotalHits?: number
+  placeholderSearch?: boolean
+}
+
+type InstantMeiliSearchInstance = {
+  pagination?: boolean
+  paginationTotalHits: number
+  hitsPerPage?: number
+  client: MeiliSearch
+  attributesToHighlight: string[]
+  placeholderSearch: boolean
+  parseMeiliSearchResponse: (
+    indexUid: string,
+    meiliSearchResponse: SearchResponse<any, any>,
+    params: Params
+  ) => {
+    results: Array<{
+      hits: any
+      index: string
+      hitsPerPage: number
+      facets: object | undefined
+      exhaustiveFacetsCount?: boolean
+      processingTimeMs: number
+      exhaustiveNbHits: boolean
+    }>
+  }
+
+  transformToMeiliSearchParams: (
+    params: Params
+  ) => {
+    q: any
+    facetsDistribution: any
+    facetFilters: any
+    attributesToHighlight: string[]
+    attributesToCrop: string[] | undefined
+    filters: any
+    limit: number
+  }
+  parseHits: (meiliSearchHits: any, params: Params) => any
+  paginationParams: (
+    hitsLength: number,
+    params: Params
+  ) => { nbPages: number; page: number | undefined } | undefined
+
+  search: (
+    requests: Requests
+  ) => Promise<{
+    results: Array<{
+      hits: any
+      index: string
+      hitsPerPage: number
+      facets: object | undefined
+      exhaustiveFacetsCount?: boolean | undefined
+      processingTimeMs: number
+      exhaustiveNbHits: boolean
+    }>
+  }>
+}
```

Comment:
Should these types not be exported as well?

Reply:
I didn't need them, but they might be useful to someone. Should I add the exports or should we wait until they're requested?

Reply:
No need to export them for this first PR 🙂 Let's keep it like this for now.